LPS-129699 Remove Liferay.Util.selectEntityHandler usage from roles

kresimir-coko opened the pull request:
This PR removes usages of `Liferay.Util.selectEntityHandler` inside `roles-admin-web`. I've manually tested the changes to make sure modals work as they did before.

## 🧪 Testing
1. Open the Control Panel and go to Users and Organizations
2. Create a new user
3. Click on Roles in the left navigation
4. `Select` Regular roles, and `Select` Site roles

liferay-continuous-integration commented:
CI is automatically triggering the following test suites:
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**relevant**
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**sf**

liferay-continuous-integration commented:
<h1>Build started.</h1><p>Jenkins is currently running tests in the <code>sf</code> test suite.</p><h4>Base Branch:</h4><p>Branch Name: <a href="https://github.com/liferay/liferay-portal/tree/master">master</a></p><h4>Job Summary:</h4><p>Job Link: <a href="https://test-1-21.liferay.com/job/test-portal-source-format/5589/">test-portal-source-format</a></p><p><b>For more details click <a href="https://test-1-21.liferay.com//userContent/jobs/test-portal-source-format/builds/5589/jenkins-report.html">here</a>.</b></p>

liferay-continuous-integration commented:
<h1>Build started.</h1><p>Jenkins is currently running tests in the <code>relevant</code> test suite.</p><h4>Base Branch:</h4><p>Branch Name: <a href="https://github.com/liferay/liferay-portal/tree/master">master</a></p><h4>Job Summary:</h4><p>Job Link: <a href="https://test-1-19.liferay.com/job/test-portal-acceptance-pullrequest(master)/8048/">test-portal-acceptance-pullrequest(master)</a></p><p><b>For more details click <a href="https://test-1-19.liferay.com//userContent/jobs/test-portal-acceptance-pullrequest(master)/builds/8048/jenkins-report.html">here</a>.</b></p>

liferay-continuous-integration commented:
<html><h3>:heavy_check_mark: ci:test:sf - 1 out of 1 jobs passed in 3 minutes</h3><details><summary>Click here for more details.</summary><h4>Base Branch:</h4><p>Branch Name: <a href="https://github.com/liferay/liferay-portal/tree/master">master</a><br/>Branch GIT ID: <a href="https://github.com/liferay/liferay-portal/commit/18fc48726acd97d5902e4a6bb337713866fc1cef">18fc48726acd97d5902e4a6bb337713866fc1cef</a></p><h4>Sender Branch:</h4><p>Branch Name: <a href="https://github.com/kresimir-coko/liferay-portal/tree/LPS-129699">LPS-129699</a><br/>Branch GIT ID: <a href="https://github.com/kresimir-coko/liferay-portal/commit/9b2b7f9f15e823e7b300d9b4eb87f537746d1772">9b2b7f9f15e823e7b300d9b4eb87f537746d1772</a></p>1 out of 1jobs PASSED<details><summary><strong>1 Successful Jobs:</strong></summary><ul><li><a href="https://test-1-21.liferay.com/job/test-portal-source-format/5589/">test-portal-source-format</a></li></ul></details><h5>For more details click <a href="https://test-1-21.liferay.com/userContent/jobs/test-portal-source-format/builds/5589/jenkins-report.html">here</a>.</h5></details></html>

kresimir-coko commented:
@markocikos helped me discovered that [users-admin-web/roles.jsp](https://github.com/liferay/liferay-portal/blob/master/modules/apps/users-admin/users-admin-web/src/main/resources/META-INF/resources/user/roles.jsp#L545-L642) is the one that handles the modal in question, so I'll migrate that one and send it.

liferay-continuous-integration commented:
<html><h3>:heavy_check_mark: ci:test:stable - 9 out of 9 jobs passed</h3><h3>:heavy_check_mark: ci:test:relevant - 23 out of 23 jobs passed in 2 hours 50 minutes</h3><details><summary>Click here for more details.</summary><h4>Base Branch:</h4><p>Branch Name: <a href="https://github.com/liferay/liferay-portal/tree/master">master</a><br/>Branch GIT ID: <a href="https://github.com/liferay/liferay-portal/commit/18fc48726acd97d5902e4a6bb337713866fc1cef">18fc48726acd97d5902e4a6bb337713866fc1cef</a></p><h4>Upstream Comparison:</h4><p>Branch GIT ID: <a href="https://github.com/liferay/liferay-portal/commit/18fc48726acd97d5902e4a6bb337713866fc1cef">18fc48726acd97d5902e4a6bb337713866fc1cef</a><br/>Jenkins Build URL: <a href="https://test-1-1.liferay.com/job/test-portal-acceptance-upstream-dxp(master)//1710">Acceptance Upstream DXP (master) #1710</a></p><details><summary><strong>ci:test:stable - 9 out of 9 jobs PASSED</strong></summary><details><summary><strong>9 Successful Jobs:</strong></summary><ul><li><a href="https://test-1-17.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/458270/">test-portal-acceptance-pullrequest-batch(master)/functional-smoke-tomcat90-mysql57-jdk8_stable/0</a></li><li><a href="https://test-1-8.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/429853/">test-portal-acceptance-pullrequest-batch(master)/functional-upgrade-tomcat90-mysql57-jdk8_stable/0</a></li><li><a href="https://test-1-10.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/383300/">test-portal-acceptance-pullrequest-batch(master)/js-unit-jdk8/0</a></li><li><a href="https://test-1-11.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/467253/">test-portal-acceptance-pullrequest-batch(master)/modules-integration-mysql57-jdk8_stable/0</a></li><li><a href="https://test-1-8.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/429850/">test-portal-acceptance-pullrequest-batch(master)/modules-unit-jdk8_stable/0</a></li><li><a href="https://test-1-11.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/467254/">test-portal-acceptance-pullrequest-batch(master)/pmd-jdk8/0</a></li><li><a href="https://test-1-4.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/470701/">test-portal-acceptance-pullrequest-batch(master)/rest-builder-jdk8/0</a></li><li><a href="https://test-1-8.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/429849/">test-portal-acceptance-pullrequest-batch(master)/service-builder-jdk8/0</a></li><li><a href="https://test-1-8.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/429852/">test-portal-acceptance-pullrequest-batch(master)/unit-jdk8_stable/0</a></li></ul></details></details><details><summary><strong>ci:test:relevant - 23 out of 23 jobs PASSED</strong></summary><details><summary><strong>23 Successful Jobs:</strong></summary><ul><li><a href="https://test-1-19.liferay.com/job/test-portal-acceptance-pullrequest(master)/8048/">test-portal-acceptance-pullrequest(master)</a></li><li><a href="https://test-1-10.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/383299/">test-portal-acceptance-pullrequest-batch(master)/central-requirements-jdk8/0</a></li><li><a href="https://test-1-17.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/458272/">test-portal-acceptance-pullrequest-batch(master)/functional-smoke-tomcat90-mysql57-jdk11_open/0</a></li><li><a href="https://test-1-17.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/458270/">test-portal-acceptance-pullrequest-batch(master)/functional-smoke-tomcat90-mysql57-jdk8_stable/0</a></li><li><a href="https://test-1-4.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/470702/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/0</a></li><li><a href="https://test-1-8.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/429855/">test-portal-acceptance-pullrequest-batch(master)/functional-upgrade-tomcat90-mysql57-jdk8/0</a></li><li><a href="https://test-1-8.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/429853/">test-portal-acceptance-pullrequest-batch(master)/functional-upgrade-tomcat90-mysql57-jdk8_stable/0</a></li><li><a href="https://test-1-10.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/383300/">test-portal-acceptance-pullrequest-batch(master)/js-unit-jdk8/0</a></li><li><a href="https://test-1-17.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/458274/">test-portal-acceptance-pullrequest-batch(master)/lpkg-base-jdk8/0</a></li><li><a href="https://test-1-17.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/458273/">test-portal-acceptance-pullrequest-batch(master)/lpkg-controller-jdk8/0</a></li><li><a href="https://test-1-17.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/458271/">test-portal-acceptance-pullrequest-batch(master)/modules-compile-jdk8/0</a></li><li><a href="https://test-1-8.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/429854/">test-portal-acceptance-pullrequest-batch(master)/modules-integration-mysql57-jdk8/0</a></li><li><a href="https://test-1-11.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/467253/">test-portal-acceptance-pullrequest-batch(master)/modules-integration-mysql57-jdk8_stable/0</a></li><li><a href="https://test-1-8.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/429851/">test-portal-acceptance-pullrequest-batch(master)/modules-semantic-versioning-jdk8/0</a></li><li><a href="https://test-1-17.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/458275/">test-portal-acceptance-pullrequest-batch(master)/modules-unit-jdk8/0</a></li><li><a href="https://test-1-8.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/429850/">test-portal-acceptance-pullrequest-batch(master)/modules-unit-jdk8_stable/0</a></li><li><a href="https://test-1-4.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/470700/">test-portal-acceptance-pullrequest-batch(master)/modules-unit-project-templates-jdk8/0</a></li><li><a href="https://test-1-11.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/467254/">test-portal-acceptance-pullrequest-batch(master)/pmd-jdk8/0</a></li><li><a href="https://test-1-4.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/470701/">test-portal-acceptance-pullrequest-batch(master)/rest-builder-jdk8/0</a></li><li><a href="https://test-1-8.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/429856/">test-portal-acceptance-pullrequest-batch(master)/semantic-versioning-jdk8/0</a></li><li><a href="https://test-1-8.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/429849/">test-portal-acceptance-pullrequest-batch(master)/service-builder-jdk8/0</a></li><li><a href="https://test-1-4.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/470699/">test-portal-acceptance-pullrequest-batch(master)/unit-jdk8/0</a></li><li><a href="https://test-1-8.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/429852/">test-portal-acceptance-pullrequest-batch(master)/unit-jdk8_stable/0</a></li></ul></details></details><h5>For more details click <a href="https://test-1-19.liferay.com/userContent/jobs/test-portal-acceptance-pullrequest(master)/builds/8048/jenkins-report.html">here</a>.</h5></details></html><h5>Test bundle downloads:</h5><ul><li><a href="https://files.liferay.com/private/bundles/test-1-19/jobs/test-portal-acceptance-pullrequest(master)/builds/8048/liferay-portal-bundle-tomcat.tar.gz">liferay-portal-bundle-tomcat.tar.gz</a> (mirrors: <a href="http://mirrors.dlc.liferay.com/files.liferay.com/private/bundles/test-1-19/jobs/test-portal-acceptance-pullrequest(master)/builds/8048/liferay-portal-bundle-tomcat.tar.gz">dlc</a>, <a href="http://mirrors.lax.liferay.com/files.liferay.com/private/bundles/test-1-19/jobs/test-portal-acceptance-pullrequest(master)/builds/8048/liferay-portal-bundle-tomcat.tar.gz">lax</a>)</li></ul>

kresimir-coko commented:
Hey @markocikos can you give this a look now? 
I've replaced the `Liferay.Util.selectEntity` with `Liferay.Util.openSelectionModal` in the `users-admin-web/roles.jsp` and added the removal of the other `selectEntityHandler` from `roles-web`.

I've tested to make sure it still works. LMK if I missed something.

markocikos commented on modules/apps/users-admin/users-admin-web/src/main/resources/META-INF/resources/user/roles.jsp:
```diff
@@ -588,57 +588,49 @@ currentURLObj.setParameter("historyKey", liferayPortletResponse.getNamespace() +
-							String siteRoleEventName = liferayPortletResponse.getNamespace() + "selectSiteRole";
-							%>
+				var selectSiteRoleLink = document.querySelector(
+					'#<portlet:namespace />selectSiteRoleLink'
```
Minor SF: we can use `getElementById`.

Also, we can use `const`.

markocikos commented on modules/apps/users-admin/users-admin-web/src/main/resources/META-INF/resources/user/roles.jsp:
```diff
@@ -588,57 +588,49 @@ currentURLObj.setParameter("historyKey", liferayPortletResponse.getNamespace() +
-						}
-					);
-				});
+							url: '<%= selectSiteRoleURL.toString() %>',
+							onSelect: function (selectedItem) {
```
Minor SF: sort, `url` > `onSelect`

markocikos reviewed (commented):
@kresimir-coko Looks not horrible at all! Some minor nits inline.

jonmak08 commented:
LGTM otherwise the small SF changes @markocikos commented on.

kresimir-coko commented:
Manually forwarded to https://github.com/pei-jung/liferay-portal/pull/823
